From 70e750804a700b06f62f91411d487b88e7c85fe5 Mon Sep 17 00:00:00 2001
From: Ashok Sidipotu <ashok.sidipotu@collabora.com>
Date: Fri, 26 Nov 2021 06:18:04 +0530
Subject: [PATCH] Revert "meson: declare spa_dep and override_dependency() for
 spa and pipewire"

This reverts commit 7ab5c35cadc075bcdf44031f1f7d95352cf9a13b.

Upstream-Status: Inappropriate[meson version dependent]

---
 meson.build              |  2 +-
 spa/meson.build          |  7 -------
 src/pipewire/meson.build | 10 ++--------
 3 files changed, 3 insertions(+), 16 deletions(-)

diff --git a/meson.build b/meson.build
index 4a572bd42..4af179c87 100644
--- a/meson.build
+++ b/meson.build
@@ -1,7 +1,7 @@
 project('pipewire', ['c' ],
   version : '0.3.40',
   license : [ 'MIT', 'LGPL-2.1-or-later', 'GPL-2.0-only' ],
-  meson_version : '>= 0.54.0',
+  meson_version : '>= 0.49.0',
   default_options : [ 'warning_level=3',
                       'c_std=gnu99',
                       'b_pie=true',
diff --git a/spa/meson.build b/spa/meson.build
index 483536b54..74ee5a737 100644
--- a/spa/meson.build
+++ b/spa/meson.build
@@ -63,12 +63,6 @@ if not get_option('examples').disabled()
   subdir('examples')
 endif
 
-spa_dep = declare_dependency(
-  include_directories : [spa_inc],
-  version : spaversion,
-  variables : { 'plugindir' : meson.current_build_dir() / 'plugins' }
-)
-
 pkgconfig.generate(filebase : 'lib@0@'.format(spa_name),
   name : 'libspa',
   subdirs : spa_name,
@@ -79,4 +73,3 @@ pkgconfig.generate(filebase : 'lib@0@'.format(spa_name),
   uninstalled_variables : ['plugindir=${prefix}/spa/plugins'],
 )
 
-meson.override_dependency('lib@0@'.format(spa_name), spa_dep)
diff --git a/src/pipewire/meson.build b/src/pipewire/meson.build
index f66ce8746..5cfaba3f9 100644
--- a/src/pipewire/meson.build
+++ b/src/pipewire/meson.build
@@ -109,12 +109,8 @@ libpipewire = shared_library(pipewire_name, pipewire_sources,
 )
 
 pipewire_dep = declare_dependency(link_with : libpipewire,
-  include_directories : [pipewire_inc, configinc],
-  dependencies : [pthread_lib, atomic_dep, spa_dep],
-  variables : {
-    'moduledir' : meson.current_build_dir() / '..' / 'modules',
-    'confdatadir' : meson.current_build_dir() / '..' / 'daemon',
-  }
+  include_directories : [pipewire_inc, configinc, spa_inc],
+  dependencies : [pthread_lib, atomic_dep, ],
 )
 
 pkgconfig.generate(libpipewire,
@@ -132,6 +128,4 @@ pkgconfig.generate(libpipewire,
   ],
 )
 
-meson.override_dependency('lib@0@'.format(pipewire_name), pipewire_dep)
-
 subdir('extensions')
-- 
2.33.1

